eth/tracers/live.go

```diff
@@ -5,9 +5,15 @@ import (
 	"errors"
 
 	"github.com/ethereum/go-ethereum/core/tracing"
+	"github.com/ethereum/go-ethereum/rpc"
 )
 
-type ctorFunc func(config json.RawMessage) (*tracing.Hooks, error)
+// LiveApiRegister is the interface that used to register JSON-RPC APIs
+type LiveApiRegister interface {
+	RegisterAPIs(apis []rpc.API)
+}
+
+type ctorFunc func(config json.RawMessage, stack LiveApiRegister, backend tracing.Backend) (*tracing.Hooks, error)
```

Comment:
This makes me realize the whole tracing interface is in one place (tracing/hooks.go) except for the constructor. It is harder to catch backwards-incompatible changes to the interface because of this. Gonna move it out.

Reply:
yeah, agree with it, move the apis into one place(hooks.go) will make it more obvious.